src/System.Management.Automation/engine/remoting/client/RemotingProtocol2.cs

```
@@ -813,18 +813,23 @@ private void HandleReadyForDisconnect(object sender, EventArgs args)
                     // what thread this callback is made from.  If it was made from a transport
                     // callback event then a deadlock may occur when DisconnectAsync is called on
                     // that same thread.
-                    ThreadPool.QueueUserWorkItem(new WaitCallback(StartDisconnectAsync), RemoteSession);
+                    ThreadPool.QueueUserWorkItem(new WaitCallback(StartDisconnectAsync));
                 }
             }
         }
 
         /// <summary>
         /// WaitCallback method to start an asynchronous disconnect.
         /// </summary>
-        /// <param name="remoteSession"></param>
-        private void StartDisconnectAsync(object remoteSession)
+        /// <param name="state"></param>
+        private void StartDisconnectAsync(object state)
         {
-            ((ClientRemoteSession)remoteSession).DisconnectAsync();
+            var remoteSession = RemoteSession;
+            try
+            {
+                remoteSession?.DisconnectAsync();
+            }
+            catch { }
```

Please add a comment why we hide exceptions.